### DISCUSSION TITLE
Tworzenie oferty z produktem

### DISCUSSION THREAD

**Discussion by winyle-online** [allegro/allegro-api#6259] [Suggestions - resolved]
Podczas wystawiania ofert przesyłam zawsze pełne dane do utworzenia ewentualnego produktu.
Zdarza się, że produkt już istnieje, ale różni się np. kategorią lub tytuł jest inaczej zapisany.

Wówczas otrzymuję takie przykładowe błędy:

> Na podstawie przesłanych danych rozpoznany został produkt z naszej bazy. Wskazana kategoria `Synth-pop` (321961) nie jest zgodna z kategorią produktu: `Rock` (261043).

lub

> Wskazany produkt istnieje. Przesłany parametr `Tytuł` o wartości `Pieśni I Arie Operetkowe` nie pasuje do parametru produktu `Pieśni i arie operetkowe`.

Moja sugestia jest taka. 
Czy w tym błędzie dałoby się zwracać od razu `id` produktu, który rozpoznał allegro?

Wówczas można by było od razu przesłać ponownie request tworzenia oferty, ale już podając id produktu z waszego systemu, skoro i tak różni się jedynie kategorią lub słowa z tytułu są zapisane mały literami.

**Comment by hahyes** [comment]
Sprawdzałeś, czy jest zwracany nagłówek `Location`?

**Reply by winyle-online** [reply]
Sprawdzałem. Nie jest zwracany. Chyba, że jakimś cudem przeoczyłem.

Obecnie zrobiłem u siebie ponowienie próby, ale przesyłając wartości, które zwróciło api. Czyli jeśli tytuł lub wykonawca znalezionego produktu nieco się różni jedynie nazwą (np. wszystkie litery były z dużej), to po ponowieniu zapytania już mam dopasowanie 100% i bezproblemowe dodanie oferty.

**Comment by MartaNowaczyk** [comment]
Sugestia wdrożona. W takich przypadkach zwracamy już ID produktu w [metadanych](https://developer.allegro.pl/news/wzbogacamy-bledy-o-uzyteczne-metadane-8dlyVqBydHO).
